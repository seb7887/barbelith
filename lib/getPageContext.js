import { SheetsRegistry } from 'jss';
import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
import createGenerateClassName from '@material-ui/core/styles/createGenerateClassName';

// App color theme
const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
    suppressDeprecationWarnings: true,
    fontFamily: ['"Segoe UI"', 'sans-serif']
  },
  palette: {
    primary: {
      main: '#fff'
    },
    secondary: {
      main: '#000'
    },
    favoriteIcon: {
      main: '#fc766a'
    },
    type: 'light'
  }
});

function createPageContext() {
  return {
    theme,
    // this is needed in order to deduplicate the injection of css in the page
    sheetsManager: new Map(),
    // this is needed in order to inject the critical css
    sheetsRegistry: new SheetsRegistry(),
    // the standard class name generator
    generateClassName: createGenerateClassName()
  };
}

export default function getPageContext() {
  // make sure to create a new context for every server-side request so that data
  // isn't shared between connections (really really BAD)
  if (!process.browser) {
    return createPageContext();
  }

  // reuse context on the client-side
  if (!global.__INIT_MATERIAL_UI__) {
    global.__INIT_MATERIAL_UI__ = createPageContext();
  }

  return global.__INIT_MATERIAL_UI__;
}
