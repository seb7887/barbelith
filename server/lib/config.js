require('dotenv').config();

const dev = process.env.NODE_ENV !== 'production';
const port = process.env.PORT || 3000;

const config = {
  port: process.env.PORT || 3000,
  db: dev ? process.env.DEV_DB : process.env.MONGO_URI,
  url: dev ? `http://localhost:${port}` : process.env.PROD_URL,
  secret: process.env.SESSION_SECRET || 'jwtSecret'
};

module.exports = config;
