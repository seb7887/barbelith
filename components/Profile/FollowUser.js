import Button from '@material-ui/core/Button';
import { followUser, unfollowUser } from '../../lib/api';

const FollowUser = ({ isFollowing, toggleFollow }) => {
  const request = isFollowing ? unfollowUser : followUser;

  return (
    <Button
      data-testid='button'
      variant='contained'
      color={isFollowing ? 'primary' : 'secondary'}
      onClick={() => toggleFollow(request)}
    >
      {isFollowing ? 'Unfollow' : 'Follow'}
    </Button>
  );
}

export default FollowUser;
