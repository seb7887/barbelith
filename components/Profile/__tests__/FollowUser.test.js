/* eslint-env jest */
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';

import { render, fireEvent } from 'react-testing-library';
import { followUser as mockedFollowUser } from '../../../lib/api';
import { unfollowUser as mockedUnfollowUser } from '../../../lib/api';
import FollowUser from '../FollowUser';

const toggleFollow = jest.fn();

jest.mock('../../../lib/api', () => {
  return {
    followUser: jest.fn(() => Promise.resolve()),
    unfollowUser: jest.fn(() => Promise.resolve())
  };
});

afterEach(() => {
  mockedFollowUser.mockClear();
  mockedUnfollowUser.mockClear();
});

describe('<FollowUser/>', () => {
  it('renders and matches snapshot', () => {
    const { asFragment } = render(
      <FollowUser isFollowing={false} toggleFollow={toggleFollow} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a follow button if is not already following', () => {
    const { getByTestId } = render(
      <FollowUser isFollowing={false} toggleFollow={toggleFollow} />
    );

    expect(getByTestId('button').textContent).toContain('Follow');
  });

  it('renders an unfollow button if is already following', () => {
    const { getByTestId } = render(
      <FollowUser isFollowing={true} toggleFollow={toggleFollow} />
    );

    expect(getByTestId('button').textContent).toContain('Unfollow');
  });

  it('calls toggleFollow if button is clicked', () => {
    const { getByTestId } = render(
      <FollowUser isFollowing={false} toggleFollow={toggleFollow} />
    );

    fireEvent.click(getByTestId('button'));

    expect(toggleFollow).toHaveBeenCalledTimes(1);
  });
});
