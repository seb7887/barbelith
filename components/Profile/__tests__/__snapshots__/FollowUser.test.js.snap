// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FollowUser/> renders and matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedSecondary-14 MuiButton-raised-15 MuiButton-raisedSecondary-17"
    data-testid="button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label-2"
    >
      Follow
    </span>
    <span
      class="MuiTouchRipple-root-30"
    />
  </button>
</DocumentFragment>
`;
